controllers/operatorpolicy_status.go

```diff
@@ -373,6 +386,51 @@ var opGroupTooManyCond = metav1.Condition{
 	Message: "there is more than one OperatorGroup in the namespace",
 }
 
+// catalogSourceFindCond is a conditionally compliant condition with reason
+// based on the `isMissing` parameter
+func catalogSourceFindCond(isMissing bool) metav1.Condition {
+	status := metav1.ConditionFalse
+	reason := "CatalogSourcesFound"
+	message := "CatalogSource was found"
+
+	if isMissing {
+		status = metav1.ConditionTrue
+		reason = "CatalogSourcesNotFound"
+		message = "CatalogSource was not found"
+	}
+
+	return metav1.Condition{
+		Type:    "CatalogSourcesUnhealthy",
+		Status:  status,
+		Reason:  reason,
+		Message: message,
+	}
+}
+
+// catalogSourceObj returns a conditionally compliant RelatedObject with reason based on the `isMissing` parameter
+func catalogSourceObj(sub *operatorv1alpha1.Subscription, isMissing bool) policyv1.RelatedObject {
```

Comment:
I think I'd prefer if this just took a name and namespace, not a whole subscription.